File: fe/lib/models/item.dart
class Item {
  final String itemId;
  final String catalogId;
  final String name;
  final String description;
  final String? imageUrl;
  final bool owned;
  final Map<String, String> userFields;
  final DateTime createdAt;
  final DateTime updatedAt;
  
  Item({
    required this.itemId,
    required this.catalogId,
    required this.name,
    required this.description,
    this.imageUrl,
    this.owned = false,
    Map<String, String>? userFields,
    required this.createdAt,
    required this.updatedAt,
  }) : userFields = userFields ?? {};
  
  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      itemId: json['item_id'] as String,
      catalogId: json['catalog_id'] as String,
      name: json['name'] as String,
      description: json['description'] as String,
      imageUrl: json['image_url'] as String?,
      owned: json['owned'] as bool? ?? false,
      userFields: (json['user_fields'] as Map<String, dynamic>?)?.map(
        (key, value) => MapEntry(key, value.toString()),
      ) ?? {},
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'item_id': itemId,
      'catalog_id': catalogId,
      'name': name,
      'description': description,
      'image_url': imageUrl,
      'owned': owned,
      'user_fields': userFields,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
  
  Item copyWith({
    String? itemId,
    String? catalogId,
    String? name,
    String? description,
    String? imageUrl,
    bool? owned,
    Map<String, String>? userFields,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Item(
      itemId: itemId ?? this.itemId,
      catalogId: catalogId ?? this.catalogId,
      name: name ?? this.name,
      description: description ?? this.description,
      imageUrl: imageUrl ?? this.imageUrl,
      owned: owned ?? this.owned,
      userFields: userFields ?? this.userFields,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: be/catalog-api/app/schemas/item.py
from pydantic import BaseModel, Field
from typing import Dict, Optional

class ItemBase(BaseModel):
    name: str = Field(..., description="아이템명")
    description: str = Field(..., description="아이템 설명")
    image_url: Optional[str] = Field(default=None, description="이미지 URL")
    user_fields: Dict[str, str] = Field(default_factory=dict, description="사용자 정의 필드")

class ItemCreate(ItemBase):
    catalog_id: str = Field(..., description="카탈로그 ID")

class ItemUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    image_url: Optional[str] = None
    user_fields: Optional[Dict[str, str]] = None

class Item(ItemBase):
    item_id: str = Field(..., description="아이템 고유 ID")
    catalog_id: str = Field(..., description="카탈로그 ID")
    owned: bool = Field(default=False, description="현재 사용자의 보유 여부")
    created_at: str = Field(..., description="생성일")
    updated_at: str = Field(..., description="수정일")

    class Config:
        from_attributes = True
